lib/versioning/poetry/index.ts

```diff
@@ -76,59 +220,41 @@ const equals = (a: string, b: string): boolean => {
   }
 };
 
-const getMajor = (version: string): number => {
-  try {
-    return npm.getMajor(padZeroes(version));
-  } catch (err) /* istanbul ignore next */ {
-    return pep440.getMajor(version);
-  }
-};
+const getMajor = (version: string): number =>
+  npm.getMajor(poetry2semver(version));
```

Comment:
please convert to `function` declarations, so it's easier to review stack traces on unexpected failures
```suggestion
function getMajor(version: string): number {
  return npm.getMajor(poetry2semver(version));
}
```